ta/pkcs11/src/processing_symm.c

```diff
@@ -246,34 +246,71 @@ static enum pkcs11_rc pkcs2tee_load_hashed_attr(TEE_Attribute *tee_ref,
 	}
 
 	res = TEE_DigestDoFinal(handle, a_ptr, a_size, hash_ptr, hash_size);
+	TEE_FreeOperation(handle);
 	if (res) {
 		EMSG("TEE_DigestOp. failed %#"PRIx32"", tee_algo);
-		TEE_FreeOperation(handle);
 		return PKCS11_CKR_FUNCTION_FAILED;
 	}
 
 	TEE_InitRefAttribute(tee_ref, tee_id, hash_ptr, *hash_size);
 
-	TEE_FreeOperation(handle);
 	return PKCS11_CKR_OK;
 }
 
+static enum pkcs11_rc hash_secret_helper(enum pkcs11_mechanism_id mech_id,
+					 struct pkcs11_object *obj,
+					 TEE_Attribute *tee_attr,
+					 void **ctx,
+					 size_t *object_size)
+{
+	uint32_t algo = 0;
+	void *hash_ptr = NULL;
+	uint32_t hash_size = 0;
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+
+	/* Find the hash algorithm */
+	rc = hmac_to_tee_hash(&algo, mech_id);
+	if (rc)
+		return rc;
+
+	hash_size = TEE_ALG_GET_DIGEST_SIZE(algo);
+	hash_ptr = TEE_Malloc(hash_size, 0);
+	if (!hash_ptr)
+		return PKCS11_CKR_DEVICE_MEMORY;
+
+	rc = pkcs2tee_load_hashed_attr(tee_attr,
+				       TEE_ATTR_SECRET_VALUE,
+				       obj,
+				       PKCS11_CKA_VALUE,
+				       algo,
+				       hash_ptr,
+				       &hash_size);
+	if (rc) {
+		EMSG("No secret/hash error");
+		TEE_Free(hash_ptr);
+		return PKCS11_CKR_FUNCTION_FAILED;
+	}
+
+	/* Store the hashed key in extra_ctx pointer */
```

Comment:
nitpicking: could remove this inline comment and the empty line below.

Reply:
Removed